lib/rules/no-restricted-imports.js

```diff
@@ -344,6 +389,27 @@ module.exports = {
 						restrictedPathEntry.importNames;
 					const allowedImportNames =
 						restrictedPathEntry.allowImportNames;
+					const allowTypeImports =
+						restrictedPathEntry.allowTypeImports;
+
+					// Skip if this is a type-only import and it's allowed for this specific entry
+					if (
+						(node.type === "ImportDeclaration" ||
+							node.type === "TSImportEqualsDeclaration") &&
+						isTypeOnlyImport(node) &&
+						allowTypeImports
+					) {
+						return;
+					}
+
+					// Skip if this is a type-only export and it's allowed for this specific entry
+					if (
+						node.type === "ExportNamedDeclaration" &&
```

Comment:
I'm not sure about the following case, since both rules report it although I think it should pass:

```js

/* eslint no-restricted-imports: ["error", {  
  "paths": [{
    "name": "foo",
    "allowTypeImports": true
  }]
}]
*/

/* eslint @typescript-eslint/no-restricted-imports: ["error", {  
  "paths": [{
    "name": "foo",
    "allowTypeImports": true
  }]
}]
*/

export type * from "foo"; // reported by both rules
``` 

It might be a false positive in `@typescript-eslint/no-restricted-imports`?